src/Microsoft.Identity.Client/ApiConfig/Executors/ClientExecutorFactory.cs

```diff
@@ -5,16 +5,22 @@ namespace Microsoft.Identity.Client.ApiConfig.Executors
 {
     internal static class ClientExecutorFactory
     {
+        private static bool IsMatsEnabled(ClientApplicationBase clientApplicationBase)
+        {
+            return clientApplicationBase.ServiceBundle.Mats != null;
+        }
+
         public static IPublicClientApplicationExecutor CreatePublicClientExecutor(
             PublicClientApplication publicClientApplication)
         {
-            var executor = new PublicClientExecutor(publicClientApplication.ServiceBundle, publicClientApplication);
+            IPublicClientApplicationExecutor executor = new PublicClientExecutor(
+                publicClientApplication.ServiceBundle,
+                publicClientApplication);
```

Comment:
nit: put inside the if (IsMats .... ?

Reply:
no, that's not how this works.  we create the "real" executor here, and if mats is enabled then we wrap it with the Mats executor and send in the "real" executor as a member to it.  we always want to create the "real" executor so that the calls go through.

---
In reply to: [264541207](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/955#discussion_r264541207) [](ancestors = 264541207)